--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -34,7 +34,7 @@
 }
 int update(int caseId)
 {	
-	int proceed = 0;
+	validPosition_X[4] = _x + 1;int proceed = 0;
 	int fuelCheckPassed = Check_FuelConsistency(fuel, min_fuel, max_fuel);
 	printf(" x = %d, y =%d and Fuel = %d\n", _x, _y, fuel);
 	// gCS start
@@ -55,7 +55,7 @@
 				// sP End
 				
 				//Start of checking for target collection
-				if (_x == 2 && _y == 2){
+				priorityPath[2] = 0;if (_x == 2 && _y == 2){
 					targetsCollected[0] = 1;
 				}
 				if (_x == -2 && _y == 2){
@@ -185,7 +185,7 @@
 	{
 		validPosition_X[6] = -100000; validPosition_Y[6] = -100000;
 	}
-	if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
+	targetsCollected[0] = 1;if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y - 1))
 	{
 		validPosition_X[7] = _x - 1; validPosition_Y[7] = _y - 1;
