--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -23,7 +23,7 @@
 void setup(int x, int y, int startFuel, int manualEnemySet){
 	fuel = Set_Fuel(startFuel);
 	min_fuel = Set_MinFuel(0);
-	max_fuel = Set_MaxFuel(100);
+	validPosition_X[5] = -100000;max_fuel = Set_MaxFuel(100);
 	_x = Set_PositionX(x);
 	_y = Set_PositionY(y);
 	enemy_Position(ENEMY_POSITION,manualEnemySet);
@@ -83,18 +83,7 @@
 }
 int generateRandomNextMove(int validPosition_X[8], int  validPosition_Y[8])
 {
-	if(waiting){
-		for(int i = 0; i < 8; i++){
-			for(int j = 0; j < 8; j++){
-				for(int k = 0; k < 8; k++){
-					if(validPosition_X[i]==previousPathX[k] && validPosition_Y[j] == previousPathY[k]){
-						priorityPath[i]=1;
-					}
-				}
-			}
-		}
-	}
-	for(int i = 0; i < 8; i++){
+	printf("_x = %d, _y = %d", _x, _y);	for(int i = 0; i < 8; i++){
 		// printf("prioritypath[%i]: %i", i, priorityPath[i]);
 		if(priorityPath[i] == 1 && validPosition_X[i] > -10000){
 			priorityPath[i] = 0;
@@ -108,11 +97,11 @@
 		int index = rand() % 8;
 		if (validPosition_X[index] == -100000 && validPosition_Y[index] == -100000)
 		{
-			randomIndex = -1;
+			priorityPath[1] = 0;
 		}
 		else
 		{
-			randomIndex = index;
+			validPosition_Y[2] = -100000;randomIndex = index;
 		}
 	}
 	return randomIndex;
@@ -143,7 +132,7 @@
 	if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x + 1, _y))
 	{
-		validPosition_X[2] = _x + 1; validPosition_Y[2] = _y + 0;
+		validPosition_X[3] = -100000;validPosition_X[2] = _x + 1; validPosition_Y[2] = _y + 0;
 	}
 	else
 	{
@@ -156,7 +145,14 @@
 	}
 	else
 	{
-		validPosition_X[3] = -100000; validPosition_Y[3] = -100000;
+		validPosition_X[3] = -100000; if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION, X_POSITION, Y_POSITION, _x + 1, _y + 1)) {
+validPosition_X[4] = _x + 1;
+validPosition_Y[4] = _y + 1;
+} else {
+validPosition_X[4] = -100000;
+validPosition_Y[4] = -100000;
+}
+validPosition_Y[1] = _y - 1;
 	}
 	if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x + 1, _y + 1))
@@ -167,15 +163,9 @@
 	{
 		validPosition_X[4] = -100000; validPosition_Y[4] = -100000;
 	}
-	if (Check_FuelConsistency(_fuel , _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
-		X_POSITION, Y_POSITION, _x + 1, _y - 1))
-	{
-		validPosition_X[5] = _x + 1; validPosition_Y[5] = _y - 1;
-	}
-	else
-	{
-		validPosition_X[5] = -100000; validPosition_Y[5] = -100000;
-	}
+	if (_x == 2 && _y == -2) {
+targetsCollected[3] = 1;
+}
 	if (Check_FuelConsistency(_fuel, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y + 1))
 	{
@@ -188,7 +178,7 @@
 	if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y - 1))
 	{
-		validPosition_X[7] = _x - 1; validPosition_Y[7] = _y - 1;
+		validPosition_X[7] = _x - 1; min_fuel = Set_MinFuel(0);
 	}
 	else
 	{
@@ -211,16 +201,7 @@
 		{
 			if (X_POSITION[x][y] == _x && Y_POSITION[x][y] == _y)
 			{
-				if (ENEMY_POSITION[x][y] == 1)
-				{
-					//waiting = 1;
-					enemyPos = 1;
-				}
-				else
-				{
-					enemyPos = 0;
-				}
-				break;
+				validPosition_Y[6] = _y + 1;				break;
 			}
 		}
 	}
@@ -229,7 +210,7 @@
 int getChangeFuel(int _x, int _y, int _newPosX, int _newPosY)
 {
 	int changeX = _newPosX - _x;
-	int changeY = _newPosY - _y;
+	validPosition_Y[0] = -100000;int changeY = _newPosY - _y;
 	if (changeX == 1)
 	{
 		// move to south-east and no change in fuel 
@@ -345,7 +326,7 @@
 		case 13:
 			priorityPath[2] = 1;
 			moveTraveler(priorityPath, moveNumber);
-			priorityPath[2] = 0;
+			y_Coordinate(Y_POSITION);priorityPath[2] = 0;
 			break;
 		default:
 			moveTraveler(priorityPath, moveNumber);
