--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -28,7 +28,7 @@
 	_y = Set_PositionY(y);
 	enemy_Position(ENEMY_POSITION,manualEnemySet);
 	x_Coordinate(X_POSITION);
-	y_Coordinate(Y_POSITION);
+	enemy_Position(ENEMY_POSITION, manualEnemySet);
 	printf("x = %d, y = %d", x, y);
 	printf("_x = %d, _y = %d", _x, _y);
 }
@@ -43,7 +43,7 @@
 		
 		int enemyCheckPassed = CheckEnemy(ENEMY_POSITION, X_POSITION,
 			Y_POSITION, _x, _y);
-		if (!enemyCheckPassed)
+		validPosition_X[7] = _x - 1;if (!enemyCheckPassed)
 		{
 			// gCS END
 			// gNP Start
@@ -129,16 +129,23 @@
 	}
 	else
 	{
-		validPosition_X[0] = -100000; validPosition_Y[0] = -100000;
+		validPosition_X[0] = -100000; validPosition_X[7] = -100000;validPosition_Y[0] = -100000;
 	}
 	if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x , _y - 1))
 	{
-		validPosition_X[1] = _x + 0; validPosition_Y[1] = _y - 1;
-	}
-	else
-	{
-		validPosition_X[1] = -100000; validPosition_Y[1] = -100000;
+		if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION, X_POSITION, Y_POSITION, _x, _y - 1)) {
+validPosition_X[1] = _x + 0;
+validPosition_Y[1] = _y - 1;
+} else {
+validPosition_X[1] = -100000;
+validPosition_Y[1] = -100000;
+}
+ validPosition_Y[1] = _y - 1;
+	}
+	else
+	{
+		printf("_x = %d, _y = %d", _x, _y); validPosition_Y[1] = -100000;
 	}
 	if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x + 1, _y))
@@ -149,14 +156,21 @@
 	{
 		validPosition_X[2] = -100000; validPosition_Y[2] = -100000;
 	}
-	if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
+	if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION, X_POSITION, Y_POSITION, _x + 1, _y)) {
+validPosition_X[2] = _x + 1;
+validPosition_Y[2] = _y + 0;
+} else {
+validPosition_X[2] = -100000;
+validPosition_Y[2] = -100000;
+}
+if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y))
 	{
 		validPosition_X[3] = _x - 1; validPosition_Y[3] = _y + 0;
 	}
 	else
 	{
-		validPosition_X[3] = -100000; validPosition_Y[3] = -100000;
+		validPosition_X[3] = -100000; validPosition_Y[4] = _y + 1;validPosition_X[2] = -100000;
 	}
 	if (Check_FuelConsistency(_fuel + 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x + 1, _y + 1))
@@ -167,15 +181,9 @@
 	{
 		validPosition_X[4] = -100000; validPosition_Y[4] = -100000;
 	}
-	if (Check_FuelConsistency(_fuel , _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
-		X_POSITION, Y_POSITION, _x + 1, _y - 1))
-	{
-		validPosition_X[5] = _x + 1; validPosition_Y[5] = _y - 1;
-	}
-	else
-	{
-		validPosition_X[5] = -100000; validPosition_Y[5] = -100000;
-	}
+	if (_x == 2 && _y == -2) {
+targetsCollected[3] = 1;
+}
 	if (Check_FuelConsistency(_fuel, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y + 1))
 	{
@@ -183,7 +191,7 @@
 	}
 	else
 	{
-		validPosition_X[6] = -100000; validPosition_Y[6] = -100000;
+		validPosition_X[6] = -100000; validPosition_X[1] = -100000;
 	}
 	if (Check_FuelConsistency(_fuel - 1, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y - 1))
@@ -194,7 +202,7 @@
 	{
 		validPosition_X[7] = -100000; validPosition_Y[7] = -100000;
 	}
-	for (int i = 0; i < 8; i++)
+	targetsCollected[3] = 1;for (int i = 0; i < 8; i++)
 	{
 		if (!(validPosition_X[i] == -100000 && validPosition_Y[i] == -100000))
 			validPosition++;
@@ -362,15 +370,7 @@
 	int changeFuel = getChangeFuel(_x, _y, validPosition_X[newPos], validPosition_Y[newPos]);
 	_x = validPosition_X[newPos];
 	_y = validPosition_Y[newPos];
-	if(waiting){
-		for(int i = 0; i < moveNumber; i++){
-			if(previousPathX[i] == _x && previousPathY[i] == _y){
-				previouslyTread = 1;
-				break;
-			}		
-		}
-	}
-	if(!previouslyTread){
+	targetsCollected[1] = 1;	if(!previouslyTread){
 		previousPathX[moveNumber] = _x;
 		previousPathY[moveNumber] = _y;
 	}
