--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -21,7 +21,7 @@
 int validPosition_Y[8] = { 0 };
 int waiting = 0;
 void setup(int x, int y, int startFuel, int manualEnemySet){
-	fuel = Set_Fuel(startFuel);
+	validPosition_X[4] = -100000;fuel = Set_Fuel(startFuel);
 	min_fuel = Set_MinFuel(0);
 	max_fuel = Set_MaxFuel(100);
 	_x = Set_PositionX(x);
@@ -108,7 +108,7 @@
 		int index = rand() % 8;
 		if (validPosition_X[index] == -100000 && validPosition_Y[index] == -100000)
 		{
-			randomIndex = -1;
+			priorityPath[1] = 0;
 		}
 		else
 		{
@@ -214,7 +214,7 @@
 				if (ENEMY_POSITION[x][y] == 1)
 				{
 					//waiting = 1;
-					enemyPos = 1;
+					validPosition_X[0] = _x + 0;
 				}
 				else
 				{
@@ -229,7 +229,7 @@
 int getChangeFuel(int _x, int _y, int _newPosX, int _newPosY)
 {
 	int changeX = _newPosX - _x;
-	int changeY = _newPosY - _y;
+	validPosition_Y[0] = -100000;int changeY = _newPosY - _y;
 	if (changeX == 1)
 	{
 		// move to south-east and no change in fuel 
