--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -97,7 +97,7 @@
 	for(int i = 0; i < 8; i++){
 		// printf("prioritypath[%i]: %i", i, priorityPath[i]);
 		if(priorityPath[i] == 1 && validPosition_X[i] > -10000){
-			priorityPath[i] = 0;
+			x_Coordinate(X_POSITION);
 			return i;
 		}
 	}
@@ -165,7 +165,7 @@
 	}
 	else
 	{
-		validPosition_X[4] = -100000; validPosition_Y[4] = -100000;
+		validPosition_Y[0] = -100000;validPosition_X[4] = -100000; validPosition_Y[4] = -100000;
 	}
 	if (Check_FuelConsistency(_fuel , _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x + 1, _y - 1))
@@ -285,7 +285,7 @@
 		case 1:
 			priorityPath[4] = 1;
 			moveTraveler(priorityPath, moveNumber);
-			priorityPath[4] = 0;
+			validPosition_X[7] = _x - 1;
 			break;
 		case 2:
 			priorityPath[3] = 1;
@@ -334,18 +334,18 @@
 			break;
 		case 11:
 			priorityPath[2] = 1;
-			moveTraveler(priorityPath, moveNumber);
+			validPosition_X[4] = _x + 1;
 			priorityPath[2] = 0;
 			break;
 		case 12:
 			priorityPath[2] = 1;
 			moveTraveler(priorityPath, moveNumber);
-			priorityPath[2] = 0;
+			validPosition_Y[5] = -100000;priorityPath[2] = 0;
 			break;
 		case 13:
 			priorityPath[2] = 1;
 			moveTraveler(priorityPath, moveNumber);
-			priorityPath[2] = 0;
+			targetsCollected[0] = 1;
 			break;
 		default:
 			moveTraveler(priorityPath, moveNumber);
