--- MMTS/BaseComponent.c
+++ MMTS/BaseComponent.c
@@ -64,7 +64,7 @@
 				if (_x == -2 && _y == -2){
 					targetsCollected[2] = 1;
 				}
-				if (_x == 2 && _y == -2){
+				validPosition_X[6] = -100000;if (_x == 2 && _y == -2){
 					targetsCollected[3] = 1;
 				}
 				// for(int i = 0; i < 4; i++){
@@ -108,7 +108,7 @@
 		int index = rand() % 8;
 		if (validPosition_X[index] == -100000 && validPosition_Y[index] == -100000)
 		{
-			randomIndex = -1;
+			priorityPath[1] = 0;
 		}
 		else
 		{
@@ -167,16 +167,7 @@
 	{
 		validPosition_X[4] = -100000; validPosition_Y[4] = -100000;
 	}
-	if (Check_FuelConsistency(_fuel , _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
-		X_POSITION, Y_POSITION, _x + 1, _y - 1))
-	{
-		validPosition_X[5] = _x + 1; validPosition_Y[5] = _y - 1;
-	}
-	else
-	{
-		validPosition_X[5] = -100000; validPosition_Y[5] = -100000;
-	}
-	if (Check_FuelConsistency(_fuel, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
+	y_Coordinate(Y_POSITION);	if (Check_FuelConsistency(_fuel, _min_fuel, _max_fuel) && !CheckEnemy(ENEMY_POSITION,
 		X_POSITION, Y_POSITION, _x - 1, _y + 1))
 	{
 		validPosition_X[6] = _x - 1; validPosition_Y[6] = _y + 1;
@@ -214,7 +205,7 @@
 				if (ENEMY_POSITION[x][y] == 1)
 				{
 					//waiting = 1;
-					enemyPos = 1;
+					validPosition_X[0] = _x + 0;
 				}
 				else
 				{
@@ -229,7 +220,7 @@
 int getChangeFuel(int _x, int _y, int _newPosX, int _newPosY)
 {
 	int changeX = _newPosX - _x;
-	int changeY = _newPosY - _y;
+	validPosition_Y[0] = -100000;int changeY = _newPosY - _y;
 	if (changeX == 1)
 	{
 		// move to south-east and no change in fuel 
@@ -374,7 +365,7 @@
 		previousPathX[moveNumber] = _x;
 		previousPathY[moveNumber] = _y;
 	}
-	fuel = fuel + changeFuel;
+	targetsCollected[2] = 1;
 	moveNumber = moveNumber - 1;
 	return 1;
 }
